i2c/oled_i2c/README.adoc

```diff
@@ -0,0 +1,68 @@
+
+The code displays a series of tiny raspberries that scroll horizontally, in the process showing you how to initialize the display, write to the entire display, write to only a portion of the display, and configure scrolling.
+
+In contrast to the sensors in the examples library, the SSD1306 is not operated via direct reads/writes to registers. Instead, a list of versatile commands is defined (see datasheet) that allows the user to access all the capabilities of the driver. Ultimately, these commands end up writing to internal registers but the key point is that the I2C write process is now different. After sending a slave address, the data that follows can be either a command, flags to follow up a command or data to be written directly into the display's RAM. A control byte is required for each write after the slave address so that the driver knows what type of data is being sent.
+
+This display is 32 pixels high by 128 pixels wide. These 32 pixels are partitioned into 4 pages, each 8 bits in height. In RAM, This looks roughly like:
+
+----
+           | COL0 | COL1 | COL2 | COL3 | ...
+    PAGE 0 |      |      |      |      |
+    PAGE 1 |      |      |      |      |
+    PAGE 2 |      |      |      |      |
+    PAGE 3 |      |      |      |      |
+    ------------------------------------
+----
+
+Within each page, we have:
+
+----
+          | COL0 | COL1 | COL2 | COL3 | ...
+    COM 0 |      |      |      |      |
+    COM 1 |      |      |      |      |
+       :  |      |      |      |      |
+    COM 7 |      |      |      |      |
+    ------------------------------------
+----
+
+Note the difference between columns in RAM and the actual segment pins on the display. The distinction between these two is important as we can remap, or add offsets between the two in the driver itself instead of moving pixels in our frame buffer code.
+
+The driver has 3 modes of transferring the pixels in RAM to the display on clock cycle (provided that the driver is set to use its RAM content to drive the display, ie. command 0xA4 is sent). We choose horizontal addressing mode which, after setting the column address and page address registers to our desired start positions, will increment the column address register until the OLED display width is reached (127 in our case) after which the column address register will reset to its starting value and the page address is incremented. Once the page register reaches the end, it will wrap around as well. Effectively, this scans across the display from top to bottom, left to right in blocks that are 8 pixels high. When a byte is sent to be written into RAM, it sets all the rows for the current position of the column address register. So, if we send 10101010, and we are on PAGE 0 and COL1, COM0 is set to 1, COM1 is set to 0, COM2 is set to 1, and so on. Effectively, the byte is "transposed" to fill a single page's column. The datasheet has further information on this and the two other modes.
```

Comment:
Is "on clock cycle" necessary in the first sentence?

Reply:
Probably not